### PR TITLE
fix(metro-config): return type of GetTransformOptions should be partial

### PR DESCRIPTION
**Pull request by tido64**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/facebook/react-native/blob/0a3c55562b5ba43b00c21ca4d7b2cba0c8eb6206/template/metro.config.js#LL10-L15C8
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/metro-config/configTypes.d.ts [hunk 1]
```diff
@@ -33,7 +33,7 @@ export type GetTransformOptions = (
     entryPoints: ReadonlyArray<string>,
     options: GetTransformOptionsOpts,
     getDependenciesOf: (filePath: string) => Promise<string[]>,
-) => Promise<ExtraTransformOptions>;
+) => Promise<Partial<ExtraTransformOptions>>;
```
Comment (tido64) [inline types/metro-config/configTypes.d.ts:36]: @huntie I think this needs to be fixed upstream as well. It's not how it is used in the template: https://github.com/facebook/react-native/blob/0a3c55562b5ba43b00c21ca4d7b2cba0c8eb6206/template/metro.config.js#LL10-L15C8

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## metro-config ([<kbd>unpkg</kbd>](https://unpkg.com/browse/metro-config@latest/))
was missing the following properties:
1. getDefaultConfig
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 137c94002a3812f08badf553f8a80c55e8afc6de
</p>

File: types/metro-config/configTypes.d.ts (hunk 1 above)
Comment (huntie) [reply types/metro-config/configTypes.d.ts:36]: Thanks, will update now.

**Review by huntie — APPROVED** [review]
LGTM :)

**Comment by tido64** [comment]
Ready to merge

**Pull request merged**